lib/services/location_search.dart

```diff
@@ -35,41 +35,56 @@ class LocationSearch extends SearchDelegate {
             mapData.controllerDestination.text = "SGW Campus, Montreal";
             mapData.controllerStarting.text = "Current Location";
             Navigator.of(context).pop();
-            mapData.controllerDestination.text = "SGW, Montreal";
             mapData.controllerStarting.text = "Current Location";
             mapData.changeStart(mapData.getCurrentLocation);
             switch (suggestionList[index][0].toString()) {
               case "H":
                 {
-                  mapData.changeCampus('sgw');
-                  mapData.changeEnd(constants.hBuilding);
+                  if (suggestionList[index][1].toString() == "A") {
```

Comment:
This will be a good candidate for refactoring when core features are done

Reply:
agreed. we should expand the search to full text find at some point.

Reply:
Yes for sure, will open a refactoring issue on that matter.